logstash_async/transport.py

```diff
@@ -3,15 +3,155 @@
 # This software may be modified and distributed under the terms
 # of the MIT license.  See the LICENSE file for details.
 
+from abc import ABC, abstractmethod
+from time import sleep
+from typing import Union
+import json
+import random
 import socket
 import ssl
 import sys
 
+from requests.auth import HTTPBasicAuth
 import pylogbeat
+import requests
 
 from logstash_async.utils import ichunked
 
 
+class Transport(ABC):
+    """The :class:`Transport <Transport>` is the abstract base class of
+    all transport protocols.
+
+    :param host: The name of the host
+    :type host: str
+    :param port: The port number of the service
+    :type port: int
+    :param timeout: The timeout for the connection
+    :type timeout: float
+    :param codec: The codec which will be used to de/encode the data (eg. plain
+    or json)
+    :type codec: str
+    :param ssl_enable: Use TLS for the transport (Default: True)
+    :type ssl_enable: bool
+    :param ssl_verify: If True the class tries to verify the TLS certificate
+    with certifi. If you pass a string with a file location to CA certificate
+    the class tries to validate it against it. (Default: True)
+    :type ssl_verify: bool or str
+    """
+
+    def __init__(
+            self,
+            host: str,
+            port: int,
+            timeout: float,
+            codec: str,
+            ssl_enable: bool = True,
+            ssl_verify: bool = True
+    ) -> None:
+        self.host = host
+        self.port = port
+        self.timeout = timeout
+        self.codec = codec
+        self.ssl_enable = ssl_enable
+        self.ssl_verify = ssl_verify
+        super().__init__()
+
+    @property
+    def host(self) -> str:
+        """host
+        :param name: The name of the host
+        :type name: str
+        :return: The current name of the host
+        :rtype: str
+        """
+        return self.__host
+
+    @host.setter
+    def host(self, name: str) -> None:
+        self.__host = name
+
+    @property
+    def port(self) -> int:
```

Comment:
I would prefer to use properties when necessary, not always. It just clutters the code.
They can easily be added later if necessary.

Reply:
I understand your point. I thought about raising exceptions for wrong inputs but let's keep it simple. I will convert all direct variables.